homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import './App.css';
+import './components/buttons.css';
+import { sortBubbles } from "./utils/sorting";
+import { generateRandomBubbles } from "./utils/arrayGenerating";
+import { BubbleGrid } from "./components/bubbleGrid";
+import { ControlButton } from "./components/buttons";
+
+const sortInterval = 40;
+
+export interface AppState {
+    status: Status,
+    actualBubbles: number[],
+    sortingInProgress: boolean
+}
+
+export enum Status {
+    Solved = "Solved",
+    NotSolved = "Not Solved"
+}
+
+class App extends React.Component<{}, AppState> {
+    constructor(props: AppState) {
+        super(props);
+
+        this.state = {
+            status: Status.NotSolved,
+            actualBubbles: generateRandomBubbles(),
+            sortingInProgress: false,
+        };
+    }
+
+    componentDidUpdate(): void {
+        const { sortingInProgress } = this.state;
+        if (sortingInProgress) {
+            setTimeout(this.sort, sortInterval);
+        }
+    }
+
+    sortingProgressHandler = (): void => {
+        const { sortingInProgress } = this.state;
+        this.setState((prevState) => ({ ...prevState, sortingInProgress: !sortingInProgress }));
```

Comment:
You don't need `...prevState`. `setState` merges objects by default.
```ts
const { sortingInProgress } = this.state;
this.setState({ sortingInProgress: !sortingInProgress });
```
or
```ts
this.setState(({ sortingInProgress }) => ({ sortingInProgress: !sortingInProgress }));
```
Both work the same.